eclass/node.eclass

```diff
@@ -0,0 +1,49 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EXPORT_FUNCTIONS src_prepare src_compile src_install
+
+RESTRICT+=" mirror"
+SLOT="0"
+
+[ -z "${SRC_URI}" ] && SRC_URI="https://registry.npmjs.org/${PN}/-/${P}.tgz"
+
+[ -z "${NODEJS_BDEPEND}" ] && BDEPEND+=" ${NODEJS_BDEPEND}"
+[ -z "${NODEJS_RDEPEND}" ] && RDEPEND+=" ${NODEJS_RDEPEND}"
+[ -z "${NODEJS_DEPEND}" ] && DEPEND+=" ${NODEJS_DEPEND}"
+[ -z "${RDEPEND}" ] && DEPEND+=" ${RDEPEND}"
+
+DEPEND+=" net-libs/nodejs"
+BDEPEND+="
+	app-misc/jq
+	net-misc/rsync
+	sys-apps/moreutils
+"
+S="${WORKDIR}/package"
+
+node_src_prepare() {
+	#remove version constraints on dependencies
+	jq 'if .dependencies? then .dependencies[] = "*" else . end' package.json | sponge package.json || die
+	default
+}
+
+node_src_compile() {
+	#here we trick npm into believing there are no dependencies so it will not try to fetch them
+	jq 'with_entries(if .key == "dependencies" then .key = "deps" else . end)' package.json | sponge package.json || die
```

Comment:
why remove the version constraints in src_prepare if npm won't see them anyway? couldn't this be achieved with `sed -i 's/dependencies/dep/' package.json` without `app-misc/jq` or `sys-apps/moreutils`? (also should probably move this to src_prepare regardless)

Reply:
Version constraints are also checked at runtime.
I'm using jq because it's the right tool to manipulate JSON files.